R/tbl-sql.R

```diff
@@ -15,13 +15,9 @@
 tbl_sql <- function(subclass, src, from, ..., vars = NULL) {
   check_dots_used()
   check_character(vars, allow_null = TRUE)
-  from_sql <- as.sql(from, con = src$con)
-  vars <- vars %||% dbplyr_query_fields(src$con, from_sql)
 
-  # If not literal sql, must be a table identifier
-  if (!(is.ident(from) || is.sql(from) || is_schema(from) || inherits(from, "Id"))) {
-    from <- as.sql(from, con = src$con)
-  }
+  from <- as_from(from)
+  vars <- vars %||% dbplyr_query_fields(src$con, from)
```

Comment:
This is one of the examples that show how complicated the handling was before.

Reply:
`as_from()` doesn't need the connection because escaping is now handled later?

Reply:
💯